qt_ui/windows/AirWingConfigurationDialog.py

```diff
@@ -176,29 +197,49 @@ def apply(self) -> Squadron:
 
 
 class SquadronConfigurationLayout(QVBoxLayout):
+    config_changed = Signal(AircraftType)
+
     def __init__(self, squadrons: list[Squadron], theater: ConflictTheater) -> None:
         super().__init__()
         self.squadron_configs = []
+        self.theater = theater
         for squadron in squadrons:
-            squadron_config = SquadronConfigurationBox(squadron, theater)
-            self.squadron_configs.append(squadron_config)
-            self.addWidget(squadron_config)
+            self.add_squadron(squadron)
 
     def apply(self) -> list[Squadron]:
         keep_squadrons = []
         for squadron_config in self.squadron_configs:
-            if squadron_config.isChecked():
-                keep_squadrons.append(squadron_config.apply())
+            keep_squadrons.append(squadron_config.apply())
         return keep_squadrons
 
+    def remove_squadron_config(self, squadron: Squadron) -> None:
+        to_remove = None
+        for squadron_config in self.squadron_configs:
+            if squadron_config.squadron == squadron:
+                to_remove = squadron_config
```

Comment:
No need to keep searching. Move the guts of `if to_remove` here and just return after.

Probably also worth `logging.error` (or maybe an exception? I don't have a good feel for how damaging inconsistent state here would be) if no match is found.

Reply:
good point!